bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,260 @@
+            case "DOWN":
+                return COMMAND_CLOSE;
+            case "STOP":
+                return COMMAND_STOP;
+            default:
+                return command.toString();
+        }
+    }
+
+    private @Nullable String sendCommand(String cmd) {
+        String url = "http://" + config.ip + "/cmd/" + cmd;
+
+        try {
+            ContentResponse resp = httpClient.newRequest(url).method(HttpMethod.GET).send();
+            logger.trace("Response: {}", resp.getContentAsString());
+            if (resp.getStatus() == 200) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+            return resp.getContentAsString();
+        } catch (InterruptedException e) {
+            logger.error("API execution has been interrupted", e);
+            updateStatus(ThingStatus.OFFLINE);
+        } catch (TimeoutException e) {
+            logger.error("Timeout during API execution", e);
+            updateStatus(ThingStatus.OFFLINE);
+        } catch (ExecutionException e) {
+            logger.error("Exception during API execution", e);
+            updateStatus(ThingStatus.OFFLINE);
+        }
+        return null;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(VektivaSmarwiConfiguration.class);
+        logger.debug("IP address: {}", config.ip);
+
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            logger.error("Cannot start http client!", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Cannot start http client!");
+            return;
+        }
+
+        future = scheduler.scheduleWithFixedDelay(this::checkStatus, 0, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private synchronized void initializeWebSocketSession() {
+        if (config.useWebSockets) {
+            closeSession();
+            session = createSession();
+            if (session != null) {
+                logger.debug("WebSocket connected!");
+            }
+        }
+    }
+
+    private void checkStatus() {
+        String url = "http://" + config.ip + "/statusn";
+
+        try {
+            ContentResponse resp = httpClient.newRequest(url).method(HttpMethod.GET).send();
+            logger.debug("status values: {}", resp.getContentAsString());
+            if (resp.getStatus() == 200) {
+                processStatusResponse(resp.getContentAsString());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "got response code: " + resp.getStatus());
+            }
+            // reconnect web socket if not connected
+            if (config.useWebSockets && (session == null || !session.isOpen()) && getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                logger.debug("Initializing WebSocket session");
+                initializeWebSocketSession();
```

Comment:
Did I understand this correctly and the websocket session will be initialized every time the scheduler calls this `checkStatus` method? I do not think that is a good idea. One time initialization should be sufficient.

Reply:
It initializes the session only when the session is not open, so it is being initialized in the beginning and everytime the session gets broken.
LGTM